src/coreclr/src/binder/bindertracing.cpp

```diff
@@ -231,3 +271,178 @@ namespace BinderTracing
         return m_ignoreBind;
     }
 }
+
+namespace BinderTracing
+{
+    ResolutionAttemptedOperation::ResolutionAttemptedOperation(AssemblyName *assemblyName, UINT_PTR binderID, const HRESULT& hr)
+        : ResolutionAttemptedOperation(assemblyName, binderID, 0, hr)
+    { }
+
+    ResolutionAttemptedOperation::ResolutionAttemptedOperation(AssemblyName *assemblyName, INT_PTR managedALC, const HRESULT& hr)
+        : ResolutionAttemptedOperation(assemblyName, 0, managedALC, hr)
+    { }
+
+    ResolutionAttemptedOperation::ResolutionAttemptedOperation(AssemblyName *assemblyName, UINT_PTR binderID, INT_PTR managedALC, const HRESULT& hr)
+        : m_hr { hr }
+        , m_stage { Stage::NotYetStarted }
+        , m_attemptInfo { assemblyName, binderID, managedALC }
+        , m_pFoundAssembly { nullptr }
+    {
+        _ASSERTE(assemblyName != nullptr);
+
+        if (!BinderTracing::IsEnabled())
+            return;
+
+        PopulateAttemptInfo(m_attemptInfo);
+        m_populatedAttemptInfo = true;
+    }
+
+    // This function simply traces out the two stages represented by the bind result.
+    // It does not change the stage/assembly of the ResolutionAttemptedOperation class instance.
+    void ResolutionAttemptedOperation::TraceBindResult(const BindResult &bindResult)
+    {
+        if (!BinderTracing::IsEnabled())
+            return;
+
+        const BindResult::AttemptResult *attempt = bindResult.GetAttempt(true /*foundInContext*/);
+        if (attempt != nullptr)
+            TraceStage(Stage::FindInLoadContext, attempt->HResult, attempt->Assembly);
+
+        attempt = bindResult.GetAttempt(false /*foundInContext*/);
+        if (attempt != nullptr)
+            TraceStage(Stage::PlatformAssemblies, attempt->HResult, attempt->Assembly);
+    }
+
+    void ResolutionAttemptedOperation::TraceStage(Stage stage, HRESULT hr, BINDER_SPACE::Assembly *resultAssembly)
+    {
+        if (!BinderTracing::IsEnabled() || stage == Stage::NotYetStarted)
+            return;
+
+        if (!m_populatedAttemptInfo)
```

Comment:
This seems to always be true since we set it in the ctor...

Reply:
This was to handle if tracing was enabled after the class was constructed but before all the stages were done. It would also be reasonable (and make things simpler - this was the reason for the existence of the AttemptInfo struct as well) to say that if tracing wasn't enabled at construction, the class doesn't trace anything.

Reply:
Binding operation is naturally a short-lived thing - and typically pretty fast. So for "monitoring" scenarios I don't think it would matter at all. It might be a tiny little bit interesting for live-debugging scenarios, but I don't really see that as a thing. If it simplifies the code - let's not support "in the middle enable"...

Reply:
Only tracing if it was enabled at construction of this class now.